source/deploy/deployed-systems.rst

```diff
@@ -0,0 +1,199 @@
+Deployed Architecture
+=====================
+
+Let's start with an overview of some typical architectures for
+our deployed sites.
+
+Note that depending on the expected
+traffic of a site, we might run everything on one
+server, or spread services out onto different servers
+(e.g. web servers, celery workers, database master,
+cache, etc.).
+
+Operating system
+~~~~~~~~~~~~~~~~
+
+Our server operating system of choice is Ubuntu Server LTS.
+For new projects, we'll use the most recent LTS release.
+
+We won't necessarily update older projects that might be on
+an older LTS release, as long as they're still getting
+security updates and we don't need some feature that's only
+available on a newer release.
+
+System updates
+--------------
+
+We have started enabling automated updates using Ubuntu's
+support for that. For now we're only enabling them for
+security updates, not all software, and we're also
+blocking updates for Salt and other packages that can
+be problematic if upgraded without first testing that
+nothing breaks.
+
+Firewalls
+---------
+
+We use ``ufw`` to manage the firewall on our servers,
+blocking all incoming ports by default and only opening
+ports for the services that server is providing.
+
+When deploying to AWS, we can go further and set up the
+server security groups so that, for example, only the
+load balancer can connect to our web server ports.
+
+System users
+------------
+
+We create a user on each system, named for the project.  We use it to
+run our Django processes and to own files and directories that need to
+be written to at run-time.
+
+On some projects though, we'll create a separate user to run the
+processes. That user only needs write access to the media directory,
+and everything else can be read-only.
+
+Each developer who is allowed to run deploys has a user account
+on each system, authorized by ssh public key, and enabled for
+passwordless sudo.  The deploy process connects to systems using
+ssh and the developer's account, then uses sudo to make updates.
+
+We typically deploy to Ubuntu, where root by default has no password,
+and we don't set one. We also disable ssh'ing into the systems
+as root. The only way to get root access is to ssh as another
+user and then use sudo.
+
+Process management
+------------------
+
+We manage our Django processes (gunicorn, celery beat, celery
+workers) using supervisor.
```

Comment:
We still control which are installed. For example, we install `nginx` and it gets system scripts setup and it might be good to point those out in the same space here.